mlx/linalg.cpp

```diff
@@ -292,4 +306,37 @@ array cholesky(
       {a});
 }
 
+array cholesky_inverse(
+    const array& L,
+    bool upper /* = false */,
+    StreamOrDevice s /* = {} */) {
+  if (L.dtype() != float32) {
+    std::ostringstream msg;
+    msg << "[linalg::cholesky] Arrays must type float32. Received array "
+        << "with type " << L.dtype() << ".";
+    throw std::invalid_argument(msg.str());
+  }
+
+  if (L.ndim() < 2) {
+    std::ostringstream msg;
+    msg << "[linalg::cholesky] Arrays must have >= 2 dimensions. Received array "
+           "with "
+        << L.ndim() << " dimensions.";
+    throw std::invalid_argument(msg.str());
+  }
+
+  if (L.shape(-1) != L.shape(-2)) {
+    throw std::invalid_argument(
+        "[linalg::cholesky] Cholesky inverse is only defined for square "
+        "matrices.");
+  }
+
+  array L_inv = tri_inv(L, upper, Device::cpu);
```

Comment:
I would probably still pass `s` instead of `Device::cpu` here. I understand we are leaving performance on the table by doing the matmul on CPU but generally all ops dispatch on the same exact stream.

Reply:
Make sense! Good motivation to have some more of linear algebra ops on GPU in the future